api/access/server_test.go

```diff
@@ -184,6 +184,45 @@ func TestServer_GetBlockHeaderByHeight(t *testing.T) {
 	})
 }
 
+func TestServer_GetCollectionByID(t *testing.T) {
+	t.Run("nominal case", func(t *testing.T) {
+		index := mocks.BaselineReader(t)
+		index.CollectionFunc = func(cID flow.Identifier) (*flow.LightCollection, error) {
+			assert.Equal(t, mocks.GenericIdentifier(0), cID)
+
+			return mocks.GenericCollection(0), nil
+		}
+
+		s := baselineServer(t)
+		s.index = index
+
+		req := &access.GetCollectionByIDRequest{Id: mocks.ByteSlice(mocks.GenericIdentifier(0))}
+		resp, err := s.GetCollectionByID(context.Background(), req)
+
+		assert.NoError(t, err)
+
+		assert.NotNil(t, resp.Collection)
+		for i, txID := range resp.Collection.TransactionIds {
+			assert.Equal(t, mocks.ByteSlice(mocks.GenericIdentifier(i)), txID)
```

Comment:
Maybe it's my paranoid brain, but I don't think these tests should rely on our knowledge of the internals of the `mocks` package. Reading this test function, one cannot know why those transactions IDs  are verified in such a way, or why they have those values.

I would rather save the `GenericCollection(0)` to a variable and then verify the returned transaction IDs against the transactions in that variable.

Reply:
For simplicity I went with assuming that any generated entity that has an ID has the matching ID from the `GenericIdentifiers` because it makes it easy to test. I don't know if changing it to be more defensive is worth it in this case :p It would improve perf though, but once more it's not an issue at the moment 🤔

Reply:
I think that's OK, that's also what I would have done. I think test code should actually know the internals of how our mocks package works.

Reply:
I'd be happier if the tests were explicit in what they check and did not entangle and make assumptions about the content of the returned `generic` structures. But if you don't mind, I wont be pedantic :p